packages/wallet/src/TxBuilder/types.ts

```diff
@@ -173,12 +188,18 @@ export interface TxBuilder {
    * In case validations fail, it creates a `TxBodyValidationError` instead of `ValidTxBody`.
    *
    * @returns {Promise<MaybeValidTx>}
-   * - In case it is a `ValidTx`, it can be used to sign and submit the transaction. This is a snapshot of transaction.
-   *   Further changes done via TxBuilder, will not update this snapshot.
-   * - In case of it is an `InvalidTx`, it embeds a TxBodyValidationError with more details
+   * - In case it is a {@link ValidTx}, it can be used to sign and submit the transaction.
+   *   This is a snapshot of transaction. Further changes done via TxBuilder, will not update this snapshot.
+   *   Once a transaction is sucessfully submitted, the TxBuilder cannot be reused, to avoid.
+   * - In case of it is an `InvalidTx`, it embeds a TxBodyValidationError with more details.
+   * - `InvalidTx` embeds a {@link TxBuilderDisabled} error if a transaction built by this TxBuilder
+   *   was already successfully submitted.
    */
   build(): Promise<MaybeValidTx>;
 
+  /** Force enable TxBuilder to be reused, even if a transaction with this configuration was successfully submitted. */
+  forceEnable(): TxBuilder;
```

Comment:
How about offering a method that removes the no longer relevant parts of the transaction. Maybe `copy()`?

Reply:
I think the idea here is to be able to reuse the parts. Although not sure if there are any uses cases outside of tests. If we want a clean state then `reset()` could work.

Reply:
> Although not sure if there are any uses cases outside of tests

If that's the case it'd be better to remove it from the builder and implement a test util to provide similar convenience

Reply:
Agree. It makes it unnecessarily complex. I will remove it. For the moment we don't need it in tests either 😄

Reply:
Done in [d0bfbb2](https://github.com/input-output-hk/cardano-js-sdk/pull/434/commits/d0bfbb2dbae618fec7ec774c5aa6b12578e92f34)